modules/flowable-engine/src/main/java/org/flowable/engine/impl/persistence/entity/ExecutionEntityManagerImpl.java

```diff
@@ -461,8 +460,7 @@ public void deleteProcessInstance(String processInstanceId, String deleteReason,
             ExecutionEntity superExecution = processInstanceExecution.getSuperExecution();
             if (superExecution != null
                     && superExecution.getCurrentFlowElement() instanceof FlowNode
-                    && ((FlowNode) superExecution.getCurrentFlowElement()).getBehavior() instanceof SubProcessActivityBehavior) {
-                SubProcessActivityBehavior subProcessActivityBehavior = (SubProcessActivityBehavior) ((FlowNode) superExecution.getCurrentFlowElement()).getBehavior();
+                    && ((FlowNode) superExecution.getCurrentFlowElement()).getBehavior() instanceof SubProcessActivityBehavior subProcessActivityBehavior) {
```

Comment:
I think that here instead of

```java
&& superExecution.getCurrentFlowElement() instanceof FlowNode
&& ((FlowNode) superExecution.getCurrentFlowElement()).getBehavior() instanceof SubProcessActivityBehavior subProcessActivityBehavior
```

we can do

```java
&& superExecution.getCurrentFlowElement() instanceof FlowNode flowNode
&& flowNode.getBehavior() instanceof SubProcessActivityBehavior subProcessActivityBehavior
```

Reply:
@filiphr made the change in the branch; thanks